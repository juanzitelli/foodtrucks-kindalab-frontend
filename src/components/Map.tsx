import React from 'react'
import Marker from './../components/Marker';
import { RootStateOrAny, useSelector } from 'react-redux'
import { FoodTruck } from './../types/FoodTruck'
import GoogleMapReact from 'google-map-react';
import _ from 'lodash'
interface Props {

}

const Map = (props: Props) => {

	const { foodTrucks } = useSelector<RootStateOrAny, { foodTrucks: FoodTruck[] }>(state => state.foodTrucks)

	return (
		<div style={{ height: '60vh', width: '100%' }}>
			<GoogleMapReact
				bootstrapURLKeys={{ key: process.env.REACT_APP_MAPS_API_KEY as string }}
				defaultCenter={{
					lat: 37.7236612795053,
					lng: -122.43594382524
				}}
				defaultZoom={13}
			>
				{
					foodTrucks !== [] ? foodTrucks.map(foodTruck => <Marker
						lat={_.toNumber(foodTruck.latitude)}
						lng={_.toNumber(foodTruck.longitude)}
						isLocation={false}
						text={`${foodTruck.applicant}`}
					/>) : <h1>Hello</h1>
				}
			</GoogleMapReact>
		</div>
	)
}

export default Map
