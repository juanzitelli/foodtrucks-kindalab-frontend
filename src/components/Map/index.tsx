import React from 'react'
import Marker from './components/Marker/index';import GoogleMapReact from 'google-map-react';
import _ from 'lodash'
import { useTypedSelector } from '../../hooks/useTypedSelector';
import { MapContainer } from './styles/MapContainer';

const Map = () => {

	const { foodTrucks } = useTypedSelector(state => state.foodTrucks)

	return (
		<MapContainer>
			<GoogleMapReact
				bootstrapURLKeys={{ key: process.env.REACT_APP_MAPS_API_KEY as string }}
				defaultCenter={{
					lat: 37.7236612795053,
					lng: -122.43594382524
				}}
				defaultZoom={13}
			>
				{
					foodTrucks !== [] ? foodTrucks.map(foodTruck => <Marker
						lat={_.toNumber(foodTruck.latitude)}
						lng={_.toNumber(foodTruck.longitude)}
						text={`${foodTruck.applicant}`}
					/>) : <h1>Hello</h1>
				}
			</GoogleMapReact>
		</MapContainer>
	)
}

export default Map
