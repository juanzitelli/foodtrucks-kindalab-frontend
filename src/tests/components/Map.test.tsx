import React from 'react';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from "redux-thunk";
import configureStore from "redux-mock-store"
import Map from '../../components/Map';

const middlewares = [thunk];
const createMockStore = configureStore(middlewares);
const initState = {
	foodTrucks: []
};
const mockStore = createMockStore(initState)
mockStore.dispatch = jest.fn();

describe('Tests on Map component', () => {
	const wrapper = shallow(
		<Provider store={mockStore}>
			<Map  />
		</Provider>)
	test('Should match snapshot', async () => {
		expect(wrapper).toMatchSnapshot();
	})
})
