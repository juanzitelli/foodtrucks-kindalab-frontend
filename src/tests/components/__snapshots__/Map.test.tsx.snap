// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on Map component Should match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [MockFunction],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [MockFunction],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Map />
</ContextProvider>
`;
